src/logic/calculate.js

```diff
@@ -0,0 +1,128 @@
+import operate from './operate';
+
+function isNumber(item) {
+  return !!item.match(/[0-9]+/);
+}
+
+/**
+ * Given a button name and a calculator data object, return an updated
+ * calculator data object.
+ *
+ * Calculator data object contains:
+ *   total:s      the running total
+ *   next:String       the next number to be operated on with the total
+ *   operation:String  +, -, etc.
+ */
+export default function calculate(obj, buttonName) {
+  if (buttonName === 'AC') {
+    return {
+      total: 0,
+      next: null,
+      operation: null,
+    };
+  }
+
+  if (isNumber(buttonName)) {
+    if (buttonName === '0' && obj.next === '0') {
+      return {};
+    }
+    // If there is an operation, update next
+    if (obj.operation) {
+      if (obj.next) {
+        return { next: obj.next + buttonName };
+      }
+      return { next: buttonName };
+    }
+    // If there is no operation, update next and clear the value
+    if (obj.next) {
+      return {
+        next: obj.next + buttonName,
+        total: null,
+      };
+    }
+    return {
+      next: buttonName,
+      total: null,
+    };
+  }
+
```

Comment:
[OPTIONAL]

- When I tried to do something like `8 +` and then I pressed any operation again like `-`, it's better if it's changed into `8 -` removing the previous operation. But in your case, it's saying invalid which shows in the below screenshot. It's better if you work on that after that your app will be fully bug-free.

![Screenshot from 2021-09-15 02-28-34](https://user-images.githubusercontent.com/22434660/133333622-43e699eb-f807-4699-be77-1c0781c40729.png)